pkg/helpers_test.go

```diff
@@ -127,7 +128,7 @@ func TestValidateK1Folder(t *testing.T) {
 	if err != nil {
 		t.Error(err)
 	}
-	_, err = os.CreateTemp(populatedTempFolder, "test.ext")
+	_, err = os.Create(fmt.Sprintf("%s/%s", populatedTempFolder, "argocd-init-values.yaml"))
```

Comment:
`os.CreateTemp` is preferable for unit tests, `os.Create` creates files in the host machine and we will be leftovers in the host machine.

`os.CreateTemp` will create the resource, and self delete it when the test finish.

Reply:
Do we have confirmation we are migrating to `CreateTemp`, I don't think this PR is changing this behavior. Would this be a blocking change?

Reply:
there is actually no migration, `os.CreateTemp` is the current code.

In Go, the os.Create function is used to create a new file or truncate an existing one. However, when writing unit tests, it's generally a good idea to avoid modifying the file system. Instead, you can use `os.CreateTemp` function to create a temporary file that is automatically cleaned up after the test is finished. This allows you to isolate your tests and avoid any unintended side effects.

Additionally, as Go's os.Create function opens the file with mode 0666 (before umask), in general it may lead to security issues if the test is run in a shared environment, testing.TempFile creates a file with mode 0600, which is more secure.

Reply:
https://github.com/kubefirst/kubefirst/issues/1097

Reply:
> Maybe use the existing ticket to track this change: - #1097

will do

Reply:
synced with Thiago in slack as well, I'll take it from here in the ticket about